aos/image_ingestion/ingest_phosim_image.ipynb

```diff
@@ -0,0 +1,341 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Ingest phosim images into Butler (Gen2) with as-built camera geometry"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "DM-28341: \"with a few phosim images, demonstrate that they can be ingested into the butler and accessed via the butler.\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We use phosim images for R22 of lsstCam generated with updated `focalplanelayout.txt` and `segmentation.txt` for [phosim_syseng4](https://jira.lsstcorp.org/browse/DM-28557), that were repackaged with new [phosim repackager](https://jira.lsstcorp.org/browse/DM-28556). Below, we ingest raw repackaged phosim images, make calibs, and perform the ISR using the updated [ts_wep](https://jira.lsstcorp.org/browse/DM-28330) tools:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The testDir is /project/scichris/aos/ts_wep/tests\n",
+      "The dataDir is /project/scichris/aos/ts_wep/tests/tmp_k6ui6rf\n",
+      "The isrDir is /project/scichris/aos/ts_wep/tests/tmp_k6ui6rf/tmpm3uegc5g\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "from matplotlib import rcParams \n",
+    "import lsst.daf.persistence as dafPersist\n",
+    "rcParams['ytick.labelsize'] = 15\n",
+    "rcParams['xtick.labelsize'] = 15\n",
+    "rcParams['axes.labelsize'] = 20\n",
+    "rcParams['axes.linewidth'] = 2\n",
+    "rcParams['font.size'] = 15\n",
+    "rcParams['axes.titlesize'] = 18\n",
+    "\n",
+    "import tempfile\n",
+    "import unittest\n",
+    "import shutil\n",
+    "\n",
+    "from lsst.ts.wep.CamDataCollector import CamDataCollector\n",
+    "from lsst.ts.wep.CamIsrWrapper import CamIsrWrapper\n",
+    "from lsst.ts.wep.Utility import (\n",
+    "    getModulePath,\n",
+    "    runProgram,\n",
+    "    getAmpImagesFromDir\n",
+    ")\n",
+    "\n",
+    "\n",
+    "# Use the test files from ts_wep tests for comcam \n",
+    "modulePath = getModulePath()\n",
+    "\n",
+    "# Make temporary test sub-directories \n",
+    "testDir = os.path.join(modulePath, \"tests\")\n",
+    "dataDir = tempfile.TemporaryDirectory(dir=testDir)\n",
+    "isrDir = tempfile.TemporaryDirectory(dir=dataDir.name)\n",
```

Comment:
I think you don't have to make any more temporary directories other than the top level one. Anything inside the temporary directory will be wiped when you clean it up I think. It might make it easier to keep track of which directories are being created.

Reply:
I removed making `dataDir` , keeping just `isrDir` under `tests` . I added cleaning up of the temporary directories at the end with `shutil`. Cleaned up places where butler is pointed to the repo.